networkx/algorithms/tree/mst.py

```diff
@@ -23,6 +23,82 @@
 from networkx.utils import UnionFind, not_implemented_for
 
 
+@not_implemented_for('multigraph')
+def boruvka_mst_edges(G, minimum=True, weight='weight', data=True):
+    """Iterates over the edges of a minimum spanning tree as computed by
+    Borůvka's algorithm.
+
+    ``G`` is a NetworkX graph. Also, the edges must have distinct
+    weights, otherwise the edges may not form a tree.
```

Comment:
I think the algorithm can be modified to work without distinct edge weights, but it may result in a larger asymptotic time complexity.